crates/db/src/lib.rs

```diff
@@ -319,21 +326,59 @@ impl QuestsDatabase for Database {
     }
 }
 
+impl Database {
```

Comment:
I don't understand why this new impl block

Reply:
There are some internal methods there to support transactions that can't be added to the other `impl` block.

Reply:
got it